google/generativeai/types/image_types/_image_types.py

```diff
@@ -0,0 +1,332 @@
+        "(either `PIL.Image.Image` or `IPython.display.Image`).\n"
+        f"However, received an object of type: {type(image)}.\n"
+        f"Object Value: {image}"
+    )
+
+
+class CheckWatermarkResult:
+    def __init__(self, predictions):
+        self._predictions = predictions
+
+    @property
+    def decision(self):
+        return self._predictions[0]["decision"]
+
+    def __str__(self):
+        return f"CheckWatermarkResult([{{'decision': {self.decision!r}}}])"
+
+    def __bool__(self):
+        decision = self.decision
+        if decision == "ACCEPT":
+            return True
+        elif decision == "REJECT":
+            return False
+        else:
+            raise ValueError("Unrecognized result")
+
+
+def check_watermark(
+    img: Union[pathlib.Path, ImageType], model_id: str = "models/image-verification-001"
+) -> "CheckWatermarkResult":
+    """Checks if an image has a Google-AI watermark.
+
+    Args:
+        img: can be a `pathlib.Path` or a `PIL.Image.Image`, `IPythin.display.Image`, or `google.generativeai.Image`.
+        model_id: Which version of the image-verification model to send the image to.
+
+    Returns:
+
+    """
+    if isinstance(img, Image):
+        pass
+    elif isinstance(img, pathlib.Path):
+        img = Image.load_from_file(img)
+    elif IPython.display is not None and isinstance(img, IPython.display.Image):
+        img = Image(image_bytes=img.data)
+    elif PIL.Image is not None and isinstance(img, PIL.Image.Image):
+        blob = _pil_to_blob(img)
+        img = Image(image_bytes=blob.data)
+    elif isinstance(img, protos.Blob):
+        img = Image(image_bytes=img.data)
+    else:
+        raise TypeError(
+            f"Not implemented: Could not convert a {type(img)} into `Image`\n    {img=}"
+        )
+
+    prediction_client = client.get_default_prediction_client()
+    if not model_id.startswith("models/"):
+        model_id = f"models/{model_id}"
+
+    instance = {"image": {"bytesBase64Encoded": base64.b64encode(img._loaded_bytes).decode()}}
+    parameters = {"watermarkVerification": True}
+
+    response = prediction_client.predict(
+        model=model_id, instances=[instance], parameters=parameters
+    )
+
+    return CheckWatermarkResult(response.predictions)
+
+
+class Image:
+    """Image."""
+
+    __module__ = "vertexai.vision_models"
+
+    _loaded_bytes: Optional[bytes] = None
+    _loaded_image: Optional["PIL.Image.Image"] = None
+
+    def __init__(
+        self,
+        image_bytes: Optional[bytes],
+    ):
+        """Creates an `Image` object.
+
+        Args:
+            image_bytes: Image file bytes. Image can be in PNG or JPEG format.
+        """
+        self._image_bytes = image_bytes
+
+    @staticmethod
+    def load_from_file(location: os.PathLike) -> "Image":
+        """Loads image from local file or Google Cloud Storage.
+
+        Args:
+            location: Local path or Google Cloud Storage uri from where to load
```

Comment:
> Google Cloud Storage uri

Does that work in this API too?

(edit: I see this was already here. please ignore)